src/bidiMapper/domains/input/InputSource.ts

```diff
@@ -0,0 +1,153 @@
+/**
+ * Copyright 2023 Google LLC.
+ * Copyright (c) Microsoft Corporation.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {Input} from '../../../protocol/protocol.js';
+
+export import SourceType = Input.SourceActionsType;
+
+export class NoneSource {
+  type = SourceType.None as const;
+}
+export class KeySource {
```

Comment:
It would be nice to have a comment here, where this constants came from

Reply:
Which constants?

Reply:
return (this.#modifiers & 2) === 2;